modules/claims_api/spec/requests/v1/disability_compensation_request_spec.rb

```diff
@@ -143,6 +143,140 @@
+                  addressChangeType: 'TEMPORARY',
+                  addressLine1: '1234 Couch Street',
+                  city: 'New York City',
+                  state: 'NY',
+                  type: 'DOMESTIC',
+                  zipFirstFive: '12345',
+                  country: 'USA'
+                }
+              end
+
+              it 'raises an exception that beginningDate is not valid' do
+                with_okta_user(scopes) do |auth_header|
+                  VCR.use_cassette('evss/claims/claims') do
+                    VCR.use_cassette('evss/reference_data/get_intake_sites') do
+                      VCR.use_cassette('evss/reference_data/countries') do
+                        par = json_data
+                        par['data']['attributes']['veteran']['changeOfAddress'] = change_of_address
+
+                        post path, params: par.to_json, headers: headers.merge(auth_header)
+                        expect(response.status).to eq(400)
+                      end
+                    end
+                  end
+                end
+              end
+            end
+          end
+        end
+
+        context 'when addressChangeType is PERMANENT' do
+          let(:change_of_address) do
+            {
+              beginningDate: (Time.zone.now + 1.month).to_date.to_s,
+              endingDate: (Time.zone.now + 2.months).to_date.to_s,
+              addressChangeType: 'PERMANENT',
+              addressLine1: '1234 Couch Street',
+              city: 'New York City',
+              state: 'NY',
+              type: 'DOMESTIC',
+              zipFirstFive: '12345',
+              country: 'USA'
+            }
+          end
+
+          it 'raises an exception that endingDate is provided but should not be' do
+            with_okta_user(scopes) do |auth_header|
+              VCR.use_cassette('evss/claims/claims') do
+                VCR.use_cassette('evss/reference_data/get_intake_sites') do
+                  VCR.use_cassette('evss/reference_data/countries') do
+                    par = json_data
+                    par['data']['attributes']['veteran']['changeOfAddress'] = change_of_address
+
+                    post path, params: par.to_json, headers: headers.merge(auth_header)
+                    expect(response.status).to eq(400)
+                  end
+                end
+              end
+            end
+          end
+        end
+
+        context 'when an invalid country is submitted' do
+          let(:change_of_address) do
+            {
+              beginningDate: (Time.zone.now + 1.month).to_date.to_s,
+              addressChangeType: 'PERMANENT',
+              addressLine1: '1234 Couch Street',
+              city: 'New York City',
+              state: 'NY',
+              type: 'DOMESTIC',
+              zipFirstFive: '12345',
+              country: 'BlahBlahBlah'
+            }
+          end
+
+          it 'raises an exception that endingDate is provided but should not be' do
```

Comment:
Probably want this spec name to be something related to the `country` attribute being invalid.

Reply:
done
https://github.com/department-of-veterans-affairs/vets-api/pull/7711/commits/68c9fe826a7eb327303ddcf51e4a12541d3d6ad3